src/Core/Geographic/Extent.js

```diff
@@ -14,513 +14,574 @@ const CARDINAL = {
-        this._values[CARDINAL.EAST] = v1;
-        this._values[CARDINAL.SOUTH] = v2;
-        this._values[CARDINAL.NORTH] = v3;
+    /**
+     * Return true if this bounding box intersect with the bouding box parameter
+     * @param {Extent} extent
+     * @returns {Boolean}
+     */
+    intersectsExtent(extent) {
+        const other = extent.crs == this.crs ? extent : extent.as(this.crs, _extent);
+        return !(this.west >= other.east ||
+                 this.east <= other.west ||
+                 this.south >= other.north ||
+                 this.north <= other.south);
     }
-    return this;
-};
 
-Extent.prototype.copy = function copy(extent) {
-    this._crs = extent._crs;
-    if (this.isTiledCrs()) {
-        this.set(extent.zoom, extent.row, extent.col);
-    } else {
-        this.set(extent._values[0], extent._values[1], extent._values[2], extent._values[3]);
+    /**
+     * Return the intersection of this extent with another one
+     * @param {Extent} extent
+     * @returns {Boolean}
+     */
+    intersect(extent) {
+        if (!this.intersectsExtent(extent)) {
+            return new Extent(this.crs, 0, 0, 0, 0);
+        }
+        if (extent.crs != this.crs) {
+            extent = extent.as(this.crs, _extent);
+        }
+        return new Extent(this.crs,
+            Math.max(this.west, extent.west),
+            Math.min(this.east, extent.east),
+            Math.max(this.south, extent.south),
+            Math.min(this.north, extent.north));
     }
-    return this;
-};
-
-Extent.prototype.union = function union(extent) {
-    if (extent.crs() != this.crs()) {
-        throw new Error('unsupported union between 2 diff crs');
+    /**
+     * Set west, east, south and north values.
+     * Or if tiled extent, set zoom, row and column values
+     * @param {number} v0 west or zoom value
+     * @param {number} v1 east or row value
+     * @param {number} v2 south or column value
+     * @param {number} v3 north value
+     * @return {Extent}
+     */
+    set(v0, v1, v2, v3) {
+        if (this.isTiledCrs()) {
+            this.zoom = v0;
+            this.row = v1;
+            this.col = v2;
+        } else {
+            this._values[CARDINAL.WEST] = v0;
+            this._values[CARDINAL.EAST] = v1;
+            this._values[CARDINAL.SOUTH] = v2;
+            this._values[CARDINAL.NORTH] = v3;
+        }
+        return this;
     }
-    const west = extent.west();
-    if (west < this.west()) {
-        this._values[CARDINAL.WEST] = west;
+    /**
+     * Copy to this extent to input extent.
+     * @param {Extent} extent
+     * @return {Extent} copied extent
+     */
+    copy(extent) {
+        this._crs = extent._crs;
+        if (this.isTiledCrs()) {
+            this.set(extent.zoom, extent.row, extent.col);
+        } else {
+            this.set(extent._values[0], extent._values[1], extent._values[2], extent._values[3]);
+        }
+        return this;
     }
+    /**
```

Comment:
Could you add a new line before the documentation ? There is multiple occurrence of this.

Reply:
yes